server/routes/lecturesRoute.js

```diff
@@ -206,4 +208,51 @@ router.get('/lectures/:lecture_id', (req, res) => {
   res.status(200).json({ id: 1, course_id: "01SQNOV" });
 });*/
 
+
+
+
+
+ /**
+ * @swagger
+ * /teachers/{teacher_id}/lectures/{lecture_id}:
+ *  delete:
+ *    tags:
+ *      - lectures
+ *    summary: "Delete a lecture"
+ *    description: "Use to request a deletion for a specific lecture"
+ *    consumes:
+ *       - "application/json"
+ *    produces:
+ *       - "application/json"
+ *    responses:
+ *       "200":
+ *         description: "Successful deletion"
+ *       "304":
+ *         description: "The lecture doesn't exist or has been deleted yet"
+ *       "400":
+ *         description: "Invalid status value"
+ *    security:
+ *     - petstore_auth:
+ *       - "write:pets"
+ *       - "read:pets"
+ */
+router.delete('/lectures/:lecture_id', authorize(role.Teacher), async (req,res)=>{
+    const teacher=req.user.sub
+    const lecture_id= +req.params.lecture_id;
+    const datetime= moment().format('YYYY-MM-DD HH:mm');
+    const lecture={datetime:datetime,lecture_id:lecture_id,teacher:teacher}
+    try{
+        let number = await lectureService.deleteLecture(lecture);
+        if(number===1)
+            return res.status(200).json({});
+        else if (number===0) 
+            return res.status(304).json({});
+            return number;
```

Comment:
In which case is this value returned? Also it is not a valid express return statement